File: moji-frontend/src/constants/welfareType.js
// constants/welfareType.js
//
// 백엔드 welfareType enum(CENTRAL/LOCAL/PRIVATE/SEOUL) 4종의 한글 라벨 매핑.
// 백엔드는 enum 상수명만 반환하므로 화면 표시용 라벨은 프론트가 관리한다.
// 출처 뱃지(WelfareCard), 검색 필터의 ChipGroup, 상세 페이지 등에서 재사용.

/**
 * welfareType enum → 한글 라벨
 * CATEGORY_REFERENCE.md / API_SPEC.md 의 welfareType 열거값과 정확히 일치해야 한다.
 */
export const WELFARE_TYPE_LABEL = {
  CENTRAL: "중앙부처",
  LOCAL: "지자체",
  PRIVATE: "민간",
  SEOUL: "서울시",
};

/**
 * UI 정렬 순서. ChipGroup·셀렉트 옵션을 만들 때 사용.
 * 중앙 → 지자체 → 민간 → 서울 순서가 사용자에게 가장 익숙하다는 가정.
 */
export const WELFARE_TYPE_ORDER = ["CENTRAL", "LOCAL", "PRIVATE", "SEOUL"];

File: moji-backend/src/main/java/com/mozi/backend/domain/welfare/entity/WelfareType.java
package com.mozi.backend.domain.welfare.entity;

/**
 * 복지 정보의 출처를 구분하는 enum.
 *
 * 4가지 크롤링 출처(복지로 중앙/지자체/민간 + 서울복지포털)별로 데이터 스키마가
 * 달라 JPA JOINED 상속 매핑의 Discriminator 값으로 사용한다. WelfareCommon 부모
 * 테이블의 welfare_type 컬럼에 STRING으로 기록되며, 각 자식 엔티티는
 * @DiscriminatorValue로 자신의 enum 이름을 선언한다.
 */
public enum WelfareType {
    CENTRAL,
    LOCAL,
    PRIVATE,
    SEOUL
}
// 이 enum의 역할: 복지 출처 식별 + JPA Discriminator 값.
// EnumType.STRING으로 매핑되며 DB에는 "CENTRAL"/"LOCAL"/"PRIVATE"/"SEOUL" 문자열로 저장.
